lib/Translation/ExportVerilog/ExportVerilog.cpp

```diff
@@ -205,25 +222,44 @@ class VerilogEmitterBase {
 
 } // end anonymous namespace
 
-void VerilogEmitterBase::emitTypePaddedToWidth(Type type, size_t padToWidth,
-                                               Operation *op) {
-  int bitWidth = getBitWidthOrSentinel(type);
-  assert(bitWidth != 0 && "Shouldn't emit zero bit declarations");
+/// Emit a type's packed dimensions, returning the number of characters emitted.
+static size_t emitTypeDims(Type type, llvm::raw_ostream &os, Operation *op) {
+  if (auto inout = type.dyn_cast<rtl::InOutType>()) {
+    return emitTypeDims(inout.getElementType(), os, op);
+  } else if (auto uarray = type.dyn_cast<rtl::UnpackedArrayType>()) {
+    return emitTypeDims(uarray.getElementType(), os, op);
+  }
+
+  size_t emittedWidth = 0;
+  int width;
+  if (auto arrayType = type.dyn_cast<rtl::ArrayType>()) {
+    emittedWidth += emitTypeDims(arrayType.getElementType(), os, op);
+    width = arrayType.getSize();
+  } else {
+    width = getBitWidthOrSentinel(type);
+    assert(width != 0 && "Shouldn't emit zero bit declarations");
+  }
 
-  if (bitWidth == -1) {
-    emitError(op, "value has an unsupported verilog type ") << type;
+  if (width == -1) {
+    op->emitError("value has an unsupported verilog type ") << type;
     os << "<<invalid type>>";
-    return;
+    return 16;
```

Comment:
I think this should return 1 or 0 to avoid returning a length longer than the measured length.

Reply:
I thought it'd be nice to align text to include the `<<invalid type>>`.